File: android-app/feature/feature-discover/src/main/java/tv/bayit/plus/feature/discover/walkthrough/WalkthroughSession.kt
package tv.bayit.plus.feature.discover.walkthrough

import tv.bayit.plus.feature.discover.model.DiscoverFeature
import java.util.UUID

data class WalkthroughSession(
    val featureId: String,
    val sessionToken: String,
    val feature: DiscoverFeature,
) {
    companion object {
        fun create(feature: DiscoverFeature): WalkthroughSession =
            WalkthroughSession(
                featureId = feature.id,
                sessionToken = UUID.randomUUID().toString(),
                feature = feature,
            )
    }
}

File: ios-app/Packages/BayitCore/Sources/BayitCore/Discover/WalkthroughSession.swift
import Foundation

public struct WalkthroughSession: Sendable {
    public let featureId: String
    public let sessionToken: String
    public let stateMachine: WalkthroughStateMachine

    public init(feature: DiscoverFeature) {
        featureId = feature.id
        sessionToken = UUID().uuidString
        stateMachine = WalkthroughStateMachine(feature: feature)
    }
}
